src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -22,13 +22,22 @@
 class EmailValidator extends ConstraintValidator
 {
     /**
-     * @var bool
+     * @var string
      */
-    private $isStrict;
+    private $defaultProfile;
 
-    public function __construct($strict = false)
+    /**
+     * @param bool   $strict  Deprecated. If the constraint does not define a
+     *                        validation profile, this will determine if
+     *                        rfc-no-warn should be used as the default profile.
+     * @param string $profile If the constraint does not define a validation
+     *                        profile, this will specify which profile to use.
+     */
+    public function __construct($strict = false, $profile = Email::PROFILE_BASIC_REGX)
```

Comment:
$profile should default to `null`, use $strict only if $profile is `null`. so strict can be ignored, if profile is given.
```php
public function __construct($strict = false, $profile = null)
{
    $this->isStrict = $strict;
    $this->defaultProfile = $profile === null
        ? $strict
            ? Email::PROFILE_RFC_DISALLOW_WARNINGS
            : Email::PROFILE_BASIC_REGX
        : $profile;
}
```